### PR TITLE
Make Microsoft.AspNetCore.Mvc.TestCommon a regular project

### PR DESCRIPTION
**Pull request by pranavkm**
* This allows specifying all it's dependencies rather than for consuming projects to do this
* Remove unused APIs
* Fix weird downgrade warnings that show up due to P2P references

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs [hunk 1]
```diff
@@ -27,8 +27,7 @@ public void ClientRulesWithCompareAttribute_ErrorMessageUsesDisplayName_WithoutL
             var adapter = new CompareAttributeAdapter(attribute, stringLocalizer: null);
 
             // Mono issue - https://github.com/aspnet/External/issues/19
-            var expectedMessage = PlatformNormalizer.NormalizeContent(
-                    "'MyPropertyDisplayName' and 'OtherPropertyDisplayName' do not match.");
+            var expectedMessage = "'MyPropertyDisplayName' and 'OtherPropertyDisplayName' do not match.";
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs]: BTW this part of the change  seems unrelated and unnecessary. You can make TestCommon a regular project without removing `PlatformNormalizer.NormalizeContent()`.

File: test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json [hunk 2]
```diff
@@ -5,11 +5,9 @@
   "dependencies": {
     "dotnet-test-xunit": "2.2.0-*",
     "Microsoft.AspNetCore.Http": "1.2.0-*",
-    "Microsoft.AspNetCore.Mvc": "1.2.0-*",
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json]: Nice this mondo dependency can be removed in this test project. Have you tried all the rest?

File: test/Microsoft.AspNetCore.Mvc.TestCommon/project.json [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+    "warningsAsErrors": true,
+    "keyFile": "../../tools/Key.snk"
+  },
   "frameworks": {
     "net451": {},
-    "netstandard1.3": {
+    "netstandard1.6": {
       "dependencies": {
-        "NETStandard.Library": "1.6.1-*"
+        "System.ComponentModel.Annotations": "4.3.0"
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.TestCommon/project.json]: Curious: Are you trying to minimize the dependencies? Or did depending on `NETStandard.Library` _cause_ a problem?

File: test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/CompareAttributeAdapterTest.cs]: FYI PR was #5196 but decisions were made offline.

File: test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json [hunk 4]
```diff
@@ -6,11 +6,10 @@
   "dependencies": {
     "dotnet-test-xunit": "2.2.0-*",
     "Microsoft.AspNetCore.Mvc": "1.2.0-*",
-    "Microsoft.AspNetCore.Mvc.Formatters.Xml": "1.2.0-*",
     "Microsoft.AspNetCore.Mvc.TestCommon": {
-      "target": "project",
-      "type": "build"
+      "target": "project"
     },
+    "Microsoft.AspNetCore.Mvc.ViewFeatures": "1.2.0-*",
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json]: Remove; `TestCommon` depends on this.

File: test/Microsoft.AspNetCore.Mvc.Formatters.Xml.Test/project.json [hunk 5]
```diff
@@ -9,8 +9,7 @@
     "Microsoft.AspNetCore.Mvc": "1.2.0-*",
     "Microsoft.AspNetCore.Mvc.Formatters.Xml": "1.2.0-*",
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Formatters.Xml.Test/project.json]: Remove; `TestCommon` depends on this.

File: test/Microsoft.AspNetCore.Mvc.IntegrationTests/project.json [hunk 6]
```diff
@@ -8,8 +8,7 @@
     "Microsoft.AspNetCore.Mvc": "1.2.0-*",
     "Microsoft.AspNetCore.Mvc.Formatters.Xml": "1.2.0-*",
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.IntegrationTests/project.json]: Remove; `TestCommon` depends on this.

File: test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json (hunk 2 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json]: A lot of projects do `AddMvc()` to test the service collection setup making it hard to remove.

File: test/Microsoft.AspNetCore.Mvc.TestCommon/project.json (hunk 3 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.TestCommon/project.json]: This was just necessary. Think we use some DataAnnotationAttributes that required this.

File: test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json (hunk 2 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Cors.Test/project.json]: :ok:

File: test/Microsoft.AspNetCore.Mvc.TestCommon/project.json (hunk 3 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.TestCommon/project.json]: :ok:
